packages/accounts-password/password_server.js

```diff
@@ -500,11 +500,16 @@ Accounts.setPassword = function (userId, newPlaintextPassword, options) {
 Meteor.methods({forgotPassword: function (options) {
   check(options, {email: String});
 
-  var user = Meteor.users.findOne({"emails.address": options.email});
+  var user = Accounts._findUserByQuery({email: options.email});
   if (!user)
     throw new Meteor.Error(403, "User not found");
 
-  Accounts.sendResetPasswordEmail(user._id, options.email);
+  const emails = _.pluck(user.emails || [], 'address');
+  const caseSensitiveEmail = _.find(emails, email => {
```

Comment:
We've had some long discussions about this... The outcome was that we're not treating stored usernames and email addresses as fully case insensitive, but do want to expose some convenience functions on the client. I agree drawing the line there is kind of arbitrary and can be confusing.

Not going all the way with case insensitivity is mostly for backwards compatibility (i.e. multiple existing users with a username or email only differing by case). But in addition, it turns out the local part of email addresses is actually case sensitive according to the RFC, although that is probably highly theoretical.

In server code, you can use `findUserByUsername` and `findUserByEmail` to do a case insensitive lookup manually, but we hadn't considered the need to know which actual email address matched to pass it on to other functions, and as you found out that now requires you to filter for the right email yourself.

The alternative would be to make all `send<...>Email` functions use the same case insensitive logic, but I'm not sure that is worth it at this point.